src/data/set/intervals/basic.lean

```diff
@@ -506,7 +506,43 @@ variables {α : Type u} [order_bot α] {a : α}
 end order_bot
 
 section linear_order
-variables {α : Type u} [linear_order α] {a a₁ a₂ b b₁ b₂ : α}
+variables {α : Type u} [linear_order α] {a a₁ a₂ b b₁ b₂ c d : α}
+
+lemma not_mem_Ici_of_lt (ha : c < a) : c ∉ Ici a :=
+by rwa [mem_Ici, not_le]
+
+lemma not_mem_Iic_of_gt (hb : b < c) : c ∉ Iic b :=
+by rwa [mem_Iic, not_le]
+
+lemma not_mem_Icc_of_lt (ha : c < a) : c ∉ Icc a b :=
+not_mem_subset Icc_subset_Ici_self (not_mem_Ici_of_lt ha)
+
+lemma not_mem_Icc_of_gt (hb : b < c) : c ∉ Icc a b :=
+not_mem_subset Icc_subset_Iic_self (not_mem_Iic_of_gt hb)
+
+lemma not_mem_Ico_of_lt (ha : c < a) : c ∉ Ico a b :=
+not_mem_subset Ico_subset_Icc_self (not_mem_Icc_of_lt ha)
+
+lemma not_mem_Ioc_of_gt (hb : b < c) : c ∉ Ioc a b :=
+not_mem_subset Ioc_subset_Icc_self (not_mem_Icc_of_gt hb)
+
+lemma not_mem_Ioi_of_le (ha : c ≤ a) : c ∉ Ioi a :=
+by rwa [mem_Ioi, not_lt]
+
+lemma not_mem_Iio_of_ge (hb : b ≤ c) : c ∉ Iio b :=
+by rwa [mem_Iio, not_lt]
+
+lemma not_mem_Ioc_of_le (ha : c ≤ a) : c ∉ Ioc a b :=
+not_mem_subset Ioc_subset_Ioi_self (not_mem_Ioi_of_le ha)
+
+lemma not_mem_Ico_of_ge (hb : b ≤ c) : c ∉ Ico a b :=
+not_mem_subset Ico_subset_Iio_self (not_mem_Iio_of_ge hb)
+
+lemma not_mem_Ioo_of_le (ha : c ≤ a) : c ∉ Ioo a b :=
+not_mem_subset Ioo_subset_Ioc_self (not_mem_Ioc_of_le ha)
+
+lemma not_mem_Ioo_of_ge (hb : b ≤ c) : c ∉ Ioo a b :=
+not_mem_subset Ioo_subset_Ico_self (not_mem_Ico_of_ge hb)
```

Comment:
Should these be `iff`s?

Reply:
The ones containing `Ici/Iic/Ioi/Iio` can be made into `iff`s but not the others.